sdk/src/Core/Amazon.Runtime/Pipeline/Handlers/BaseAuthResolverHandler.cs

```diff
@@ -56,17 +76,22 @@ protected void PreInvoke(IExecutionContext executionContext)
                     continue;
                 }
 
-                // TODO: Retrieve identity resolver and signer for the current scheme and attach them to request context.
-                // Interfaces haven't been defined yet, but code will be similar to:
-                // var identityResolver = scheme.IdentityResolver();
-                // var identity = identityResolver.GetIdentity();
-                // var signer = scheme.Signer();
+                executionContext.RequestContext.Identity = this.Identity;
+                if (executionContext.RequestContext.Identity == null)
+                {
+                    var identityResolver = scheme.GetIdentityResolver(executionContext.RequestContext.ClientConfig.IdentityResolverConfiguration);
+                    executionContext.RequestContext.Identity = identityResolver.ResolveIdentity();
+                }
 
-                // This code would also need to handle existing scenarios, such as:
-                // - Customer explicitly set credentials on their service client (i.e. new AmazonS3Client(new BasicAWSCredentials))
+                executionContext.RequestContext.Signer = GetSigner(scheme);
             }
         }
 
+        protected virtual AbstractAWSSigner GetSigner(IAuthScheme<BaseIdentity> scheme)
```

Comment:
Should this return `ISigner`. `AbstractAWSSigner` seems tired to a SigV4(a) signer.

Reply:
All the signers do implement `AbstractAWSSigner`, and this is the type of `executionContext.RequestContext.Signer` 
so we will have to cast it again anyway.
![image](https://github.com/user-attachments/assets/4480d664-e6d1-4293-9b7f-bae429fab46b)

Changing `executionContext.RequestContext.Signer` to `ISigner` wont be very simple since we do depend on signer referencing core internal types that need to be extracted / refactored (e.g. IRequest).